### ISSUE TITLE
tsplot raises exception when DataFrame index doesn't start at 0

### ISSUE THREAD

**Issue by jgosmann** [mwaskom/seaborn#957]
Given this data frame:

``` python
df = pd.DataFrame({'t': np.arange(10), 'unit': 1, 'y': np.sin(np.arange(10)), 'q': ([0] * 5) + ([1] * 5)})
```

we can plot a time series with `tsplot` like so:

``` python
sns.tsplot(time='t', unit='unit', value='y', data=df)
```

If we, however, query a subset of the data frame (with indices not starting at 0) with

``` python
sns.tsplot(time='t', unit='unit', value='y', data=df.query("q == 1"))
```

an exception is raised:

```
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-31-37f9eda2cc86> in <module>()
----> 1 sns.tsplot(time='t', unit='unit', value='y', data=df.query("q == 1"))

/Volumes/Home/blubb/Library/Python/2.7/lib/python/site-packages/seaborn/timeseries.pyc in tsplot(data, time, unit, condition, value, err_style, ci, interpolate, color, estimator, n_boot, err_palette, err_kws, legend, ax, **kwargs)
    327 
    328     # Pad the sides of the plot only when not interpolating
--> 329     ax.set_xlim(x.min(), x.max())
    330     x_diff = x[1] - x[0]
    331     if not interpolate:

UnboundLocalError: local variable 'x' referenced before assignment
```

The reason is that the result of the `query` retains the original indices, but the way [the default `condition` argument is created](https://github.com/mwaskom/seaborn/blob/master/seaborn/timeseries.py#L184) assigns indices starting at 0 which then [eads to an empty number of groups in the main plotting loop](https://github.com/mwaskom/seaborn/blob/master/seaborn/timeseries.py#L277).

As a workaround it is possible to use

``` python
sns.tsplot(time='t', unit='unit', value='y', data=df.query("q == 1").reset_index())
```

**Comment by dreyco676** [comment]
Has this been fixed in a newer release? I'm on seaborn 0.7.1 with Python 2.7.11 :: Anaconda 2.3.0 (64-bit)

**Comment by mwaskom** [comment]
Actually I think it has: https://github.com/mwaskom/seaborn/commit/a2c4602c9569c17d256dae97c25aa69da63c48ef

**Comment by jgosmann** [comment]
Seems to be fixed on master (but not 0.7.1). :)

**Comment by srodriguex** [comment]
Yes.

Please, refer to [link](https://gist.github.com/srodriguex/520f7c816b9a80d516999d6dc846d142) to open the Notebook that demonstrates the problem. The Notebook contains an example from `hypertools` library that works fine with some data, and my use with my data that generates the error.

Looking around it seems the variable `x` in line 331 of timeseries.py file really looks referenced before an assignment. The assign of `x` variable occurs in line 280, inside a for.

Thanks for your support!
